src/main/java/net/pms/dlna/DVDISOTitle.java

```diff
@@ -18,75 +18,124 @@
 import net.pms.io.OutputParams;
 import net.pms.io.ProcessWrapperImpl;
 import net.pms.util.FileUtil;
+import net.pms.util.Iso639;
+import net.pms.util.MPlayerDvdAudioStreamChannels;
+import net.pms.util.MPlayerDvdAudioStreamTypes;
 import net.pms.util.ProcessUtil;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class DVDISOTitle extends DLNAResource {
 	private static final Logger LOGGER = LoggerFactory.getLogger(DVDISOTitle.class);
+
+	public static final Pattern AUDIO_STREAM_PATTERN = Pattern.compile(
+		"^audio stream: (?<StreamNumber>\\d+) format: (?<Codec>\\S+) \\((?<Channels>\\S+)\\) language: (?<Language>\\w*) aid: (?<AID>\\d+)\\.$"
+	);
+	public static final Pattern SUBTITLE_STREAM_PATTERN = Pattern.compile(
+		"^subtitle \\( sid \\): (?<StreamNumber>\\d+) language: (?<Language>\\w*)$"
+	);
+
+
 	private File file;
 	private int title;
 	private long length;
+	private String parentName;
 
+	/**
+	 * @deprecated Use {@link #DVDISOTitle(File, String, int)} instead.
+	 */
+	@Deprecated
 	public DVDISOTitle(File file, int title) {
+		this(file, null, title);
+	}
+
+	public DVDISOTitle(File file, String parentName, int title) {
 		this.file = file;
 		this.title = title;
+		this.parentName = parentName;
 		setLastModified(file.lastModified());
 	}
 
 	@Override
 	protected void resolveOnce() {
-		String cmd[] = new String[] {
-			configuration.getMplayerPath(),
-			"-identify",
-			"-endpos",
-			"0",
-			"-v",
-			"-ao",
-			"null",
-			"-vc",
-			"null",
-			"-vo",
-			"null",
-			"-dvd-device",
-			ProcessUtil.getShortFileNameIfWideChars(file.getAbsolutePath()),
-			"dvd://" + title
-		};
+		if (getMedia() == null) {
+			setMedia(new DLNAMediaInfo());
+		}
 
 		OutputParams params = new OutputParams(configuration);
 		params.maxBufferSize = 1;
+		params.log = true;
+
+		boolean generateThumbnails = false;
 		if (configuration.isDvdIsoThumbnails()) {
 			try {
 				params.workDir = configuration.getTempFolder();
+				generateThumbnails = true;
 			} catch (IOException e1) {
-				LOGGER.debug("Caught exception", e1);
+				LOGGER.error("Could not create temporary folder, DVD thumbnails won't be generated: {}", e1.getMessage());
+				LOGGER.trace("", e1);
 			}
-			cmd[2] = "-frames";
-			cmd[3] = "2";
-			cmd[7] = "-quiet";
-			cmd[8] = "-quiet";
-			String frameName = "" + this.hashCode();
-			frameName = "mplayer_thumbs:subdirs=\"" + frameName + "\"";
-			frameName = frameName.replace(',', '_');
-			cmd[10] = "jpeg:outdir=" + frameName;
 		}
-		params.log = true;
+
+		String cmd[];
+		if (generateThumbnails) {
+			String outFolder = "jpeg:outdir=mplayer_thumbs:subdirs=\"" + this.hashCode() + "\"";
+			cmd = new String[] {
+				configuration.getMplayerPath(),
+				"-identify",
+				"-ss",
+				Integer.toString(configuration.getThumbnailSeekPos()),
+				"-frames",
+				"1",
+				"-v",
+				"-ao",
+				"null",
+				"-vo",
+				outFolder,
```

Comment:
@Sami32 Yes, we don't want audio to be played, so we're setting the driver to "null". That's documented somewhere, and it's also how UMS did it before.